search_test.go

```diff
@@ -183,3 +184,81 @@ func TestUnmarshalingSearchResult(t *testing.T) {
 		t.Errorf("expected 1 error, got %d", len(rv.Status.Errors))
 	}
 }
+
+func TestFacetNumericDateRangeRequests(t *testing.T) {
+	var drMissingErr = fmt.Errorf("date range query must specify either start, end or both for range name 'testName'")
+	var nrMissingErr = fmt.Errorf("numeric range query must specify either min, max or both for range name 'testName'")
+	var drNrErr = fmt.Errorf("facet can only conain numeric ranges or date ranges, not both")
+	value := float64(5)
+
+	facet := NewFacetRequest("Date_Range_Success_With_StartEnd", 1)
+	facet.DateTimeRanges = make([]*dateTimeRange, 0, 1)
+	facet.DateTimeRanges = append(facet.DateTimeRanges, &dateTimeRange{Name: "testName", Start: time.Unix(0, 0), End: time.Now()})
+	if err := facet.Validate(); err != nil {
```

Comment:
As discussed in the meeting, instead of making repeated calls to Validate(), we should use the Go test pattern of defining test cases up front, with slice of anonymous structs, like:

https://github.com/couchbase/go-couchbase/blob/master/vbmap.go#L70-L77

And then have a for loop over those test cases like:
https://github.com/blevesearch/bleve/blob/master/search_test.go#L72